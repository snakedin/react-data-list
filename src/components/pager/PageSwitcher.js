import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withOptions } from "../../options/withOptions";

class PageSwitcher extends Component {

    static propTypes = {
        page: PropTypes.number,
        lastPage: PropTypes.bool,
        onParamChanged: PropTypes.func
    };

    /**
     * Render previous page button
     */
    getPreviousButton() {

        const { page, onParamChanged } = this.props;
        const label = (
            <span aria-hidden="true">&laquo;</span>
        );

        if (page <= 1) {
            return (
                <li className="page-item disabled" key="prev">
                    <button className="page-link" tabIndex="-1">{ label }</button>
                </li>
            )
        } else {
            return (
                <li className="page-item" key="prev">
                    <button className="page-link" aria-label="Previous" onClick={ () => { onParamChanged('page', page-1) } }>{ label }</button>
                </li>
            );
        }
    };

    /**
     * Render next page button
     */
    getNextButton() {

        const { page, lastPage, onParamChanged } = this.props;
        const label = (
            <span aria-hidden="true">&raquo;</span>
        );

        if (lastPage === true) {
            return (
                <li className="page-item disabled" key="next">
                    <button className="page-link" tabIndex="-1">{ label }</button>
                </li>
            )
        } else {
            return (
                <li className="page-item" key="next">
                    <button className="page-link" aria-label="Next" onClick={ () => { onParamChanged('page', page+1) } }>{ label }</button>
                </li>
            );
        }
    };

    /**
     * Render element
     */
    render() {

        const { page, lastPage,  options: { classNamePager } } = this.props;

        if (page === 1 && lastPage === true) {
            return null;
        }

        return (
            <ul className={ classNamePager }>
                { this.getPreviousButton() }
                { this.getNextButton() }
            </ul>
        );
    }
}

export default withOptions(PageSwitcher);