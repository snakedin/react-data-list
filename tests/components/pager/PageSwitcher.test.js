import React from "react";
import { render } from "@testing-library/react";
import "../../mocks/withOptionsMock";
import PageSwitcher from "../../../src/components/pager/PageSwitcher";

it("pager/PageSwitcher render", () => {
  const { rerender, asFragment } = render(<PageSwitcher page={1} lastPage={false} />);
  expect(asFragment()).toMatchSnapshot();

  rerender(<PageSwitcher page={2} lastPage={false} />);
  expect(asFragment()).toMatchSnapshot();

  rerender(<PageSwitcher page={3} lastPage={true} />);
  expect(asFragment()).toMatchSnapshot();
});
