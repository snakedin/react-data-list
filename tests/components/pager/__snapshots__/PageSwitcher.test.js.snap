// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pager/PageSwitcher render 1`] = `
<DocumentFragment>
  <ul
    class="pagination mb-0 justify-content-center justify-content-lg-start"
  >
    <li
      class="page-item disabled"
    >
      <button
        class="page-link"
        tabindex="-1"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
      </button>
    </li>
    <li
      class="page-item"
    >
      <button
        aria-label="Next"
        class="page-link"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`pager/PageSwitcher render 2`] = `
<DocumentFragment>
  <ul
    class="pagination mb-0 justify-content-center justify-content-lg-start"
  >
    <li
      class="page-item"
    >
      <button
        aria-label="Previous"
        class="page-link"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
      </button>
    </li>
    <li
      class="page-item"
    >
      <button
        aria-label="Next"
        class="page-link"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`pager/PageSwitcher render 3`] = `
<DocumentFragment>
  <ul
    class="pagination mb-0 justify-content-center justify-content-lg-start"
  >
    <li
      class="page-item"
    >
      <button
        aria-label="Previous"
        class="page-link"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
      </button>
    </li>
    <li
      class="page-item disabled"
    >
      <button
        class="page-link"
        tabindex="-1"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
